src/commands/Utility/spotify.js

```diff
@@ -1,95 +1,136 @@
+const RichEmbed = require('discord.js').RichEmbed;
+const SpotifyWebApi = require('spotify-web-api-node');
+const SpotifyUriParser = require('spotify-uri');
 const got = require('got');
+const spotifyApi = new SpotifyWebApi();
+
+exports.run = async function (bot, msg, args) {
+    await got('https://spotify-auth.doxylamin.pw/').then((response) => {
+        spotifyApi.setAccessToken(response.body);
+    });
 
-exports.run = function (bot, msg, args) {
     if (args.length < 1) {
-        throw 'You must specify a track url (spotify:track:XYZ)';
+        throw 'You must specify a spotify uri at least!';
     }
 
-    // possible inputs:
-    // https://open.spotify.com/track/4SUBEjh0WcaPXNeBpuRC7a
-    // spotify:track:4SUBEjh0WcaPXNeBpuRC7a
-    // 4SUBEjh0WcaPXNeBpuRC7a
+    let parsed = bot.utils.parseArgs(args, ['preview']);
+    let spotifyUri = SpotifyUriParser.parse(parsed.leftover.join(' '));
 
-    let input = args.join(' ')
-        .replace('https://open.spotify.com/track/', '')
-        .replace('spotify:track:', '');
-    let url = `https://api.spotify.com/v1/tracks/${input}`;
-    msg.edit(':arrows_counterclockwise:  Loading track info for ' + input);
+    switch(spotifyUri.type) {
+    case 'track':
+        getTrackEmbed(msg, spotifyUri.id, parsed.options.preview);
+        break;
+    case 'artist':
+        getArtistEmbed(msg, spotifyUri.id, parsed.options.preview);
+        break;
+    case 'playlist':
+        getPlaylistEmbed(msg, spotifyUri.user, spotifyUri.id, parsed.options.preview);
+        break;
+    default:
+        throw 'Sorry, I can\'t parse that type of URI yet.';
+    }
 
-    got(url).then(res => {
+};
 
-        let data;
+function getPlaylistEmbed(msg, user, spotifyId, withPreview) {
+    spotifyApi.getPlaylist(user, spotifyId)
+    .then(function(data) {
```

Comment:
Please use lambda notation (`(...) => { ... }`) instead of function notation (`function (...) { ... }`) for the sake of consistency in code style.

Reply:
Just as an example, in this case, you would do `.then(data => { ... })` (no parenthesis required because it only has a single argument for the lambda)